File: Back/.Net/BetFriends.Domain/Members/Member.cs
using BetFriends.Domain.Bets;
using BetFriends.Domain.Friendships;
using BetFriends.Domain.Members.Exceptions;

namespace BetFriends.Domain.Members;

public class Member : Entity
{
    private readonly MemberId memberId;
    private readonly string username;
    private readonly int chips;
    private readonly int countFriends;

    public Member(MemberId memberId, string username, int chips, int countFriends)
    {
        this.memberId = memberId;
        this.username = username;
        this.chips = chips;
        this.countFriends = countFriends;
    }

    public MemberId MemberId { get => memberId; }

    public Friendship AddFriendship(Guid requesterId)
    {
        return Friendship.Create(memberId.Value, requesterId);
    }

    public Bet Bet(BetId betId,
                   string description,
                   int chips,
                   DateTime endDate,
                   IEnumerable<Guid> friends)
    {
        if (this.chips < chips)
            throw new CannotBetException(CannotBetException.ChipsNotEnough);

        if (!HasFriend())
            throw new CannotBetException(CannotBetException.NoneFriends);
        return Bets.Bet.Create(betId,
                               memberId,
                               description,
                               chips,
                               endDate,
                               friends);
    }

    private bool HasFriend()
    {
        return countFriends != 0;
    }
}

File: Back/NodeJS/src/domain/members/Member.ts
import { MemberId } from './MemberId'
import { Friendship } from '../friendships/Friendship';
import { Bet } from '../bets/Bet';
import { BetId } from '../bets/BetId';
import { NotEnoughChipsException } from './exceptions/NotEnoughChipsException';
import { NoneFriendException } from './exceptions/NoneFriendException';
import { AnswerBet } from '../answerBets/AnswerBet';

export class Member {
    
    Bet(betId: string, description: string, chips: number, endDate: Date, members: string[]): Bet {
        if(this.Coins < chips) {
            throw new NotEnoughChipsException();
        }

        if(this.CountFriend == 0) {
            throw new NoneFriendException();
        }

        return Bet.Create(new BetId(betId),
                        this.MemberId,
                        description,
                        chips,
                        endDate,
                        members);
    }

    DecreaseBalance(chips: number) {
        this.Coins -= chips; 
    }

    constructor(public readonly MemberId: MemberId, 
                public readonly Username: string,
                public Coins: number, 
                public CountFriend: number){}

    AddFriendship(requesterId: string) {
        return Friendship.Create(new MemberId(requesterId), this.MemberId);
    }
    
    RejectBet(betId: BetId) {
        return new AnswerBet(betId, false, this.MemberId);
    }

    AcceptBet(betId: BetId) {
        return new AnswerBet(betId, true, this.MemberId);
    }
}